CMakeLists.txt

```diff
@@ -204,6 +223,10 @@ option(BUILD_CLBLAST_BENCHMARKS "Whether to build clBLAST benchmarks" OFF)
 option(BUILD_CLBLAS_BENCHMARKS "Whether to build clBLAS benchmarks" OFF)
 option(BUILD_ACL_BENCHMARKS "Whether to build ARM Compute Library benchmarks" OFF)
 option(BLAS_BUILD_SAMPLES "Whether to build SYCL-BLAS samples" ON)
+if (INSTALL_HEADER_ONLY)
+  set(BLAS_ENABLE_BENCHMARK OFF)
+  set(BLAS_BUILD_SAMPLES OFF)
+endif()
```

Comment:
I think it would be better to create an error here if these are on instead.

Reply:
I find it annoying to have to set 3 flags to off when you turn on  `INSTALL_HEADER_ONLY`. I've added a message instead, I hope that's enough.